### PR TITLE
develop: add playground gitpod

### PR DESCRIPTION
**Pull request by shczhen**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/playground-gitpod.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+![](/media/develop/playground-gitpod-workspace-ready.png)
+
+您可以按照 [该指南](/develop/sample-application-spring-boot.md#第-6-步http-请求) 测试 API。 请记住将 URL `http://localhost:8080` 替换为您在 `REMOTE EXPLORER` 中找到的那个。
+
+## 使用自定义的 Gitpod 配置和 Docker 镜像
+
+### 自定义 Gitpod 配置
+
+Gitpod 工作空间通过一个 `.gitpod.yml` 文件进行配置，该文件位于项目的根目录，用 YAML 语法编写。 这是一个示例：[示例 .gitpod.yml](https://github.com/shczhen/tidb-example-java/blob/main/.gitpod.yml)。
```
Comment (shczhen) [inline develop/playground-gitpod.md:45]: Need to update URL

File: develop/playground-gitpod.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    onOpen: ignore
+```
+
+### 自定义 Gitpod Docker 镜像
+
+> 默认情况下，Gitpod 使用名为 Workspace-Full 的标准 Docker 映像作为工作空间的基础。 基于此默认映像启动的工作区预装了 Docker、Go、Java、Node.js、C/C++、Python、Ruby、Rust、PHP 以及 Homebrew、Tailscale、Nginx 等工具。
+
+您可以提供公共 Docker 映像或 Dockerfile。 并为您的项目安装所需的任何依赖项。
+
+这是一个 Dockerfile 示例：[示例 .gitpod.Dockerfile](https://github.com/shczhen/tidb-example-java/blob/main/.gitpod.Dockerfile)
```
Comment (shczhen) [inline develop/playground-gitpod.md:80]: Need to update URL

File: TOC.md [hunk 3]
```diff
@@ -76,6 +76,8 @@
     - 规范
       - [命名规范](/develop/object-naming-guidelines.md)
       - [SQL 开发规范](/develop/sql-development-specification.md)
+  - 云开发环境
```
Comment (winkyao) [inline TOC.md:79]:
```suggestion
  - 云原生开发环境
```

File: develop/playground-gitpod.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    visibility: public
+  - port: 4000
+    visibility: public
+  - port: 2379-36663
+    onOpen: ignore
+```
+
+### 下一步
+
+重新启动工作区以应用更改。
```
Comment (winkyao) [inline develop/playground-gitpod.md:121]: add some example about how to change the code and rebuild de application?

File: develop/playground-gitpod.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+<!-- markdownlint-disable MD029 -->
+
+# Gitpod
+
+使用 [Gitpod](https://www.gitpod.io/) ，只需单击一个按钮/链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。
+
+> [Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务提供全新的自动化开发环境， 非常迅速。 它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
+
+## 快速开始 {#quick-start}
```
Comment (Icemap) [inline develop/playground-gitpod.md:13]: 这里的 `{#quick-start}` 将会被渲染成什么特殊样式吗？

File: develop/playground-gitpod.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+访问 `https://gitpod.io/#<YOUR_REPO_URL>` 以建立新的 Gitpod 工作区，新工作区会应用最新的代码。
+
+访问 `https://gitpod.io/workspaces` 以获取所有建立的工作区。
+
+## 总结
+
+Gitpod 提供了完整的、自动化的、预配置的云原生开发环境，无需本地配置。在浏览器中开发、运行、测试代码。
+
+![playground gitpod summary](playground-gitpod-summary.png)
```
Comment (Icemap) [inline develop/playground-gitpod.md:171]: 请将图片文件放置在 `media/develop` 下

File: develop/playground-gitpod.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
+---
+title: Gitpod
+---
+
+<!-- markdownlint-disable MD029 -->
+
+# Gitpod
+
+使用 [Gitpod](https://www.gitpod.io/) ，只需单击一个按钮/链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。
+
```
Comment (Icemap) [inline develop/playground-gitpod.md:10]:
```suggestion

> **提示：**
>
```

File: develop/playground-gitpod.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+> [Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务提供全新的自动化开发环境， 非常迅速。 它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
```
Comment (Icemap) [inline develop/playground-gitpod.md:11]:
```suggestion
> [Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务提供全新的自动化开发环境，非常迅速。它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
```

File: develop/playground-gitpod.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+
+- 例如，`https://gitpod.io/#https://github.com/pingcap-inc/tidb-example-java`。
+
+- 支持环境变量，例如，`https://gitpod.io/#targetFile=spring-jpa-hibernate_Makefile,targetMode=spring-jpa-hibernate/https://github.com/shczhen/tidb-example-java`。
+
+3. 使用列出的提供商之一登录并启动工作区，例如，`Github`。
+
+## 使用默认的 Gitpod 配置和环境
+
+在步骤 [快速开始](#quick-start) 之后，Gitpod 会需要一段时间来设置您的工作区。
```
Comment (Icemap) [inline develop/playground-gitpod.md:29]:
```suggestion
在步骤[快速开始](#quick-start)之后，Gitpod 会需要一段时间来设置您的工作区。
```

File: develop/playground-gitpod.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+这里我们使用 [Spring Boot Web](/develop/sample-application-spring-boot.md) 应用程序为例，通过 URL `https://gitpod.io/#targetFile=spring-jpa-hibernate_Makefile,targetMode=spring-jpa-hibernate/https://github.com/shczhen/tidb-example-java`创建了一个新工作区。
```
Comment (Icemap) [inline develop/playground-gitpod.md:31]:
```suggestion
这里我们使用 [Spring Boot Web](/develop/sample-application-spring-boot.md) 应用程序为例，通过 URL `https://gitpod.io/#targetFile=spring-jpa-hibernate_Makefile,targetMode=spring-jpa-hibernate/https://github.com/shczhen/tidb-example-java` 创建了一个新工作区。
```

File: develop/playground-gitpod.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+完成后，您将看到如下所示的页面。
+
+![playground gitpod workspace init](/media/develop/playground-gitpod-workspace-init.png)
+
+在这个场景中，我们使用 [TiUP](https://docs.pingcap.com/zh/tidb/stable/tiup-overview) 来搭建一个 TiDB Playground。 您可以在终端的左侧找到进度。
```
Comment (Icemap) [inline develop/playground-gitpod.md:37]:
```suggestion
在这个场景中，我们使用 [TiUP](https://docs.pingcap.com/zh/tidb/stable/tiup-overview) 来搭建一个 TiDB Playground。您可以在终端的左侧找到进度。
```

File: develop/playground-gitpod.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+一旦 TiDB Playground 准备就绪，另一个 `Spring JPA Hibernate` 任务将运行。 您可以在终端的右侧找到进度。
+
+完成所有任务后，您可以看到如下所示的页面。 您可以在左侧导航栏的`REMOTE EXPLORER`中找到您的端口 `8080` URL（Gitpod 支持基于 URL 的端口转发）。
```
Comment (Icemap) [inline develop/playground-gitpod.md:41]:
```suggestion
完成所有任务后，您可以看到如下所示的页面。您可以在左侧导航栏的 `REMOTE EXPLORER` 中找到您的端口 `8080` URL（Gitpod 支持基于 URL 的端口转发）。
```

File: develop/playground-gitpod.md [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+
+![playground gitpod workspace ready](/media/develop/playground-gitpod-workspace-ready.png)
+
+您可以按照 [该指南](/develop/sample-application-spring-boot.md#第-6-步http-请求) 测试 API。 请记住将 URL `http://localhost:8080` 替换为您在 `REMOTE EXPLORER` 中找到的那个。
```
Comment (Icemap) [inline develop/playground-gitpod.md:45]:
```suggestion
您可以按照[该指南](/develop/sample-application-spring-boot.md#第-6-步http-请求)测试 API。请记住将 URL `http://localhost:8080` 替换为您在 `REMOTE EXPLORER` 中找到的那个。
```

File: develop/playground-gitpod.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+    visibility: public
+  - port: 4000
+    visibility: public
+  - port: 2379-36663
+    onOpen: ignore
+```
+
+### 应用更改
+
+完成对`.gitpod.yml`文件配置后，请保证最新的代码已在您对应的 GitHub 代码仓库中可用。
```
Comment (Icemap) [inline develop/playground-gitpod.md:161]:
```suggestion
完成对 `.gitpod.yml` 文件配置后，请保证最新的代码已在您对应的 GitHub 代码仓库中可用。
```

File: develop/playground-gitpod.md [hunk 15]
```diff
@@ -0,0 +1,171 @@
+---
+title: Gitpod
+---
+
+<!-- markdownlint-disable MD029 -->
+
+# Gitpod
+
+使用 [Gitpod](https://www.gitpod.io/) ，只需单击一个按钮/链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。
```
Comment (Icemap) [inline develop/playground-gitpod.md:9]:
```suggestion
使用 [Gitpod](https://www.gitpod.io/)，只需单击一个按钮/链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。
```

File: develop/playground-gitpod.md (hunk 2 above)
Comment (sykp241095) [reply develop/playground-gitpod.md]: 这个是不是直接用最新的 java example url 就可以了

File: develop/playground-gitpod.md (hunk 5 above)
Comment (shczhen) [reply develop/playground-gitpod.md:13]: 针对anchor的custom-id，anchor会变成#quick-start

File: develop/playground-gitpod.md (hunk 2 above)
Comment (shczhen) [reply develop/playground-gitpod.md]: 对的

**Review by winkyao — APPROVED** [review]
LGTM

File: develop/dev-guide-playground-gitpod.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+
+<!-- markdownlint-disable MD029 -->
+
+# Gitpod
+
+> **提示：**
+>
+> 使用 [Gitpod](https://www.gitpod.io/)，只需单击一个按钮/链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。
+
+> [Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务提供全新的自动化开发环境，非常迅速。它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
```
Comment (qiancai) [inline develop/dev-guide-playground-gitpod.md:13]:
```suggestion
[Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源的 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务快速提供自动化开发环境。

使用 [Gitpod](https://www.gitpod.io/)，你只需单击一个按钮或链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。

它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
```

Comment (qiancai) [inline develop/dev-guide-playground-gitpod.md:13]:
```suggestion
使用 [Gitpod](https://www.gitpod.io/)，只需单击一个按钮或链接即可在浏览器中获得完整的开发环境，并且可以立即编写代码。

[Gitpod](https://github.com/gitpod-io/gitpod) 是一个开源 Kubernetes 应用程序，适用于可直接编写代码的开发人员环境，可为云中的每个任务提供全新的自动化开发环境，非常迅速。它使您能够将您的开发环境描述为代码，并直接从您的浏览器或桌面 IDE 启动即时、远程和基于云的开发人员环境。
```

Comment (qiancai) [reply develop/dev-guide-playground-gitpod.md:13]: 您 -> 你

File: develop/dev-guide-playground-gitpod.md [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+这是一个 TiDB 应用开发实例代码仓库 [pingcap-inc/tidb-example-java](https://github.com/pingcap-inc/tidb-example-java)。
+
+1. Fork 这个[代码仓库](https://github.com/pingcap-inc/tidb-example-java)。
+
+2. 通过浏览器的地址栏，在整个 URL 前加上 `https://gitpod.io/#` 来启动您的 gitpod 工作区。
+
+- 例如，`https://gitpod.io/#https://github.com/pingcap-inc/tidb-example-java`。
+
+- 支持环境变量，例如，`https://gitpod.io/#targetFile=spring-jpa-hibernate_Makefile,targetMode=spring-jpa-hibernate/https://github.com/pingcap-inc/tidb-example-java`。
+
```
Comment (qiancai) [inline develop/dev-guide-playground-gitpod.md:26]: 23， 25 行需要缩进

File: develop/dev-guide-playground-gitpod.md [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+完成对 `.gitpod.yml` 文件配置后，请保证最新的代码已在你对应的 GitHub 代码仓库中可用。
+
+访问 `https://gitpod.io/#<YOUR_REPO_URL>` 以建立新的 Gitpod 工作区，新工作区会应用最新的代码。
+
+访问 `https://gitpod.io/workspaces` 以获取所有建立的工作区。
+
+## 总结
+
+Gitpod 提供了完整的、自动化的、预配置的云原生开发环境，无需本地配置。在浏览器中开发、运行、测试代码。
```
Comment (qiancai) [inline develop/dev-guide-playground-gitpod.md:169]:
```suggestion
Gitpod 提供了完整的、自动化的、预配置的云原生开发环境。无需本地配置，你可以直接在浏览器中开发、运行、测试代码。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**